codegen/type_resolution.py

```diff
@@ -1038,12 +1040,27 @@ def select_function_with_least_implicit_conversion_cost(
             [(fn.format_for_output_to_user(), fn.loc) for _, fn in best_functions],
         )
 
+    def add_function_to_codegen(
+        self, declaration: FunctionDeclaration, signature: FunctionSignature
+    ) -> None:
+        for other in self._already_codegened[declaration.uuid]:
+            if (
+                do_specializations_match(other.specialization, signature.specialization)
+                and other.arguments == signature.arguments
+                and other.return_type == signature.return_type
+            ):
+                return  # Already codegened
+
+        self._already_codegened[declaration.uuid].append(signature)
+        self._remaining_to_codegen.append((declaration, signature))
+
     def lookup_function(
         self,
         name: str,
         given_specialization: list[SpecializationItem],
         args: list[TypedExpression] | list[Type],
-    ) -> FunctionSignature:
+        evaluated_context: bool,
```

Comment:
Is this always `true`?

Reply:
At the moment yes. It will only be false when we add something like `decltype`.

Reply:
Oh I see